pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -72,8 +73,27 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, pr *v1beta1.PipelineRun)
 
 	// Get TaskRun names depending on whether embeddedstatus feature is set or not
 	var trs []string
-	for _, cr := range pr.Status.ChildReferences {
-		trs = append(trs, cr.Name)
+	if len(pr.Status.ChildReferences) == 0 {
+		fieldName := "TaskRuns"
+		value := reflect.ValueOf(pr.Status)
+		field := value.FieldByName(fieldName)
```

Comment:
Why use reflection here? I think we can just revert what we did in https://github.com/tektoncd/chains/commit/e813eceaa32ee2711841ae896851b6b5cffa8b00

Reply:
Good point. I removed it.